File: sorting-searching/sorting-algorithms/counting-sort/Counting.java
import java.util.Arrays;

public class Counting {
    static void countingSort(int[] arr) {
        int max = Arrays.stream(arr).max().getAsInt();
        int[] count = new int[max + 1];

        for (int num : arr) count[num]++;

        int index = 0;
        for (int i = 0; i <= max; i++) {
            while (count[i]-- > 0) {
                arr[index++] = i;
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {4,2,2,8,3,3,1};
        countingSort(arr);
        System.out.println(Arrays.toString(arr));
    }
}

File: sorting-searching/sorting-algorithms/counting-sort/counting.js
function countingSort(arr) {
  let max = Math.max(...arr);
  let count = new Array(max + 1).fill(0);

  for (let num of arr) count[num]++;

  let index = 0;
  for (let i = 0; i <= max; i++) {
    while (count[i]-- > 0) {
      arr[index++] = i;
    }
  }

  return arr;
}

console.log(countingSort([4,2,2,8,3,3,1]));